neuraxle_tensorflow/tensorflow_v1.py

```diff
@@ -0,0 +1,215 @@
+        self.tensorflow_props = {}
+
+    def setup(self) -> BaseStep:
+        """
+        Setup tensorflow 1 graph, and session using a variable scope.
+
+        :return: self
+        :rtype: BaseStep
+        """
+        if self.is_initialized:
+            return self
+
+        self.graph = tf.Graph()
+        with self.graph.as_default():
+            with tf.variable_scope(self.variable_scope, reuse=tf.AUTO_REUSE):
+                self.session = tf.Session(config=tf.ConfigProto(log_device_placement=True), graph=self.graph)
+
+                self.tensorflow_props = self.setup_graph()
+                self.tensorflow_props = self.tensorflow_props if self.tensorflow_props is not None else {}
+
+                init = tf.global_variables_initializer()
+                self.session.run(init)
+                self.is_initialized = True
+
+    def teardown(self) -> BaseStep:
+        """
+        Close session on teardown.
+
+        :return:
+        """
+        if self.session is not None:
+            self.session.close()
+        self.is_initialized = False
+
+        return self
+
+    def strip(self):
+        """
+        Strip tensorflow 1 properties from to step to make the step serializable.
+
+        :return: stripped step
+        :rtype: BaseStep
+        """
+        self.tensorflow_props = {}
+        self.graph = None
+        self.session = None
+
+        return self
+
+    @abstractmethod
+    def setup_graph(self):
+        raise NotImplementedError()
+
+    def fit(self, data_inputs, expected_outputs=None) -> 'BaseStep':
+        with tf.variable_scope(self.variable_scope, reuse=tf.AUTO_REUSE):
+            return self.fit_model(data_inputs, expected_outputs)
+
+    @abstractmethod
+    def fit_model(self, data_inputs, expected_outputs=None) -> BaseStep:
+        """
+        Fit tensorflow model using the variable scope.
+
+        :param data_inputs: data inputs
+        :param expected_outputs: expected outputs to fit on
+        :return: fitted self
+        :rtype: BaseStep
+        """
+        raise NotImplementedError()
+
+    def transform(self, data_inputs, expected_outputs=None) -> 'BaseStep':
+        with tf.variable_scope(self.variable_scope, reuse=tf.AUTO_REUSE):
+            return self.transform_model(data_inputs)
+
+    @abstractmethod
+    def transform_model(self, data_inputs):
+        """
+        Transform tensorflow model using the variable scope.
+
+        :param data_inputs:
+        :return:
+        """
+        raise NotImplementedError()
+
+    def __getitem__(self, item):
+        """
+        Get a graph tensor by name using get item.
+
+        :param item: tensor name
+        :type item: str
+
+        :return: tensor
+        :rtype: tf.Tensor
+        """
+        if item in self.tensorflow_props:
+            return self.tensorflow_props[item]
+
+        return self.graph.get_tensor_by_name(
+            "{0}/{1}:0".format(self.variable_scope, item)
```

Comment:
0: looks like a 2nd argument

Reply:
we can't have a second argument in getitem. ill just split the string by : for the device... if : is not there in the string, ill take :0 as default

Reply:
It's possible with numpy like `matrix[a:b, c:d]`, which is here a tuple of two slices for instance. Did you look at this?
https://stackoverflow.com/a/1685412/2476920